scroll-search.lic

```diff
@@ -0,0 +1,533 @@
+    bput("open my #{cur_scroll_container}", 'That is already open', 'You open')
+
+    bput("get #{cur_scroll_ordinal} #{cur_scroll_type} from my #{cur_scroll_container}", 'You get')
+
+	# Reorder the origin container
+	reorder_container(scroll_info.container, scroll_info, true)
+	resave_container(scroll_info.container, scroll_info, true)
+	
+	container_noun = container
+	container = full_name(container)
+
+	# Just warn the user if we cannot get
+	# the full name of the container for some reason.
+	if container =~ /Undefined/i
+	  echo("Was unable to retrieve the full name of this container...") if $debug_mode
+	end
+	
+	# Reorder destination container
+	reorder_container(container, scroll_info, false)
+	
+	bput("put my #{cur_scroll_type} in my #{container_noun}", 'You put')
+	
+	# Transferred scrolls will always have a "first" ordinal
+	scroll_info.set_ordinal($ORDINALS[0])
+	scroll_info.set_container_index(0)
+	
+	# Edit scroll information
+	scroll_info.set_container(container)
+	
+	# Resave scroll info
+	resave_container(container, scroll_info, false)
+
+	echo_message("SCROLL WAS TRANSFERRED!")
+  end
+
+  def reorder_container(container, scroll_info, wasRemoved)
+	parameter = "C #{scroll_info.get_type}"
+	scrolls = rummage(parameter, get_noun(container))
+	
+	echo("Scroll count of those matching the type (#{scroll_info.get_type}) is: #{scrolls.length}.") if $debug_mode
+	
+	scrolls_in_container = @owned_scrolls[container]
+	scrolls_in_container.each do |scroll_information|
+	
+	  if scroll_information.get_type != scroll_info.get_type
+		next
+	  end
+		
+	  if scroll_information.index == scroll_info.index
+		next
+	  end
+		
+	  if wasRemoved
+		if scroll_information.container_index > scroll_info.container_index && scroll_information.container_index > 0
+		  new_index = scroll_information.container_index - 1
+		  new_ordinal = $ORDINALS[new_index]
+		else
+		  new_index = scroll_information.container_index
+		  new_ordinal = $ORDINALS[new_index]
+		end
+	  else
+		new_index = scroll_information.container_index + 1
+		new_ordinal = $ORDINALS[new_index]
+	  end
+		
+	  echo("New ordinal and index for scroll (#{scroll_information.spell}) is: #{new_index} - #{new_ordinal}.") if $debug_mode
+		
+	  scroll_information.set_ordinal(new_ordinal)
+	  scroll_information.set_container_index(new_index)
+	end
+  end
+
+  def resave_container(container, scroll_info, remove)
+  
+	# if wasRemoved is true, we remove the scroll from
+	# the container. If not, we simply add it to the container specified.
+	
+	if remove
+	  scrolls_in_container = @owned_scrolls[scroll_info.container]
+	  removed = scrolls_in_container.delete(scroll_info)
+	  if removed
+		echo("Deleting scroll #{removed.spell} from container #{scroll_info.container}") if $debug_mode
+		# Re-save owned scrolls listing
+		@owned_scrolls[scroll_info.container] = scrolls_in_container
+			
+		# Re-save user variables
+		UserVars.owned_scrolls = @owned_scrolls
+	  else
```

Comment:
I recommend checking this case first (e.g. `unless removed`) and then, due to the exit, you do not need to have an else.